File: src/main/scala/org/openapitools/models/OrdersStatsOrderDTO.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import ympa_scala_finch_server._
import java.time.LocalDateTime
import java.time.ZonedDateTime
import org.openapitools.models.OrderStatsStatusType
import org.openapitools.models.OrdersStatsCommissionDTO
import org.openapitools.models.OrdersStatsDeliveryRegionDTO
import org.openapitools.models.OrdersStatsItemDTO
import org.openapitools.models.OrdersStatsOrderPaymentType
import org.openapitools.models.OrdersStatsPaymentDTO
import scala.collection.immutable.Seq

/**
 * Информация о заказе.
 * @param id Идентификатор заказа.
 * @param creationDate Дата создания заказа.  Формат даты: `ГГГГ-ММ-ДД`. 
 * @param statusUpdateDate Дата и время, когда статус заказа был изменен в последний раз.  Формат даты и времени: ISO 8601. Например, `2017-11-21T00:00:00`. Часовой пояс — UTC+03:00 (Москва). 
 * @param status 
 * @param partnerOrderId Идентификатор заказа в информационной системе магазина.
 * @param paymentType 
 * @param fake Тип заказа:  * `false` — настоящий заказ покупателя.  * `true` — [тестовый](../../pushapi/concepts/sandbox.md) заказ Маркета. 
 * @param deliveryRegion 
 * @param items Список товаров в заказе после возможных изменений.
 * @param initialItems Список товаров в заказе до изменений.
 * @param payments Информация о денежных переводах по заказу.
 * @param commissions Информация о комиссиях за заказ.
 */
case class OrdersStatsOrderDTO(id: Option[Long],
                creationDate: Option[LocalDateTime],
                statusUpdateDate: Option[ZonedDateTime],
                status: Option[OrderStatsStatusType],
                partnerOrderId: Option[String],
                paymentType: Option[OrdersStatsOrderPaymentType],
                fake: Option[Boolean],
                deliveryRegion: Option[OrdersStatsDeliveryRegionDTO],
                items: Seq[OrdersStatsItemDTO],
                initialItems: Option[Seq[OrdersStatsItemDTO]],
                payments: Seq[OrdersStatsPaymentDTO],
                commissions: Seq[OrdersStatsCommissionDTO]
                )

object OrdersStatsOrderDTO {
    /**
     * Creates the codec for converting OrdersStatsOrderDTO from and to JSON.
     */
    implicit val decoder: Decoder[OrdersStatsOrderDTO] = deriveDecoder
    implicit val encoder: ObjectEncoder[OrdersStatsOrderDTO] = deriveEncoder
}

File: lib/app/Models/OrdersStatsOrderDTO.php
<?php
/**
 * OrdersStatsOrderDTO
 */
namespace app\Models;

/**
 * OrdersStatsOrderDTO
 * @description Информация о заказе.
 */
class OrdersStatsOrderDTO {

    /** @var int $id Идентификатор заказа.*/
    public $id = 0;

    /** @var \DateTime $creationDate Дата создания заказа.  Формат даты: &#x60;ГГГГ-ММ-ДД&#x60;.*/
    public $creationDate;

    /** @var \DateTime $statusUpdateDate Дата и время, когда статус заказа был изменен в последний раз.  Формат даты и времени: ISO 8601. Например, &#x60;2017-11-21T00:00:00&#x60;. Часовой пояс — UTC+03:00 (Москва).*/
    public $statusUpdateDate;

    /** @var string $status */
    public $status = "";

    /** @var string $partnerOrderId Идентификатор заказа в информационной системе магазина.*/
    public $partnerOrderId = "";

    /** @var string $paymentType */
    public $paymentType = "";

    /** @var bool $fake Тип заказа:  * &#x60;false&#x60; — настоящий заказ покупателя.  * &#x60;true&#x60; — [тестовый](../../pushapi/concepts/sandbox.md) заказ Маркета.*/
    public $fake = false;

    /** @var \app\Models\OrdersStatsDeliveryRegionDTO $deliveryRegion */
    public $deliveryRegion;

    /** @var \app\Models\OrdersStatsItemDTO[] $items Список товаров в заказе после возможных изменений.*/
    public $items = [];

    /** @var \app\Models\OrdersStatsItemDTO[]|null $initialItems Список товаров в заказе до изменений.*/
    public $initialItems = null;

    /** @var \app\Models\OrdersStatsPaymentDTO[] $payments Информация о денежных переводах по заказу.*/
    public $payments = [];

    /** @var \app\Models\OrdersStatsCommissionDTO[] $commissions Информация о комиссиях за заказ.*/
    public $commissions = [];

}
